sharedUiComponents/lines/color3LineComponent.tsx

```diff
@@ -17,47 +17,99 @@ export interface IColor3LineComponentProps {
     propertyName: string;
     onPropertyChangedObservable?: Observable<PropertyChangedEvent>;
     isLinear?: boolean;
-    icon? : string;
+    icon?: string;
     lockObject?: LockObject;
-    iconLabel? : string;
+    iconLabel?: string;
+    onValueChange?: (value: string) => void;
 }
 
-export class Color3LineComponent extends React.Component<IColor3LineComponentProps, { isExpanded: boolean, color: Color3 }> {
+export class Color3LineComponent extends React.Component<IColor3LineComponentProps, { isExpanded: boolean, color: Color3 | Color4, colorText: string }> {
     private _localChange = false;
     constructor(props: IColor3LineComponentProps) {
         super(props);
 
-        this.state = { isExpanded: false, color: this.props.target[this.props.propertyName].clone() };
+        const typeName = typeof (this.props.target[this.props.propertyName]);
+        if (typeName === "string") {
+            let colorConverted = this.convertToColor3(this.props.target[this.props.propertyName]);
+            this.state = { isExpanded: false, color: colorConverted, colorText: this.props.target[this.props.propertyName] };
+        } else {
+            this.state = { isExpanded: false, color: this.props.target[this.props.propertyName].clone(), colorText: this.props.target[this.props.propertyName].toHexString() };
+        }
 
         if (props.isLinear) {
             this.state.color.toGammaSpaceToRef(this.state.color);
         }
 
         props.target._isLinearColor = props.isLinear; // so that replayRecorder can append toLinearSpace() as appropriate
+        this._colorPickerOpen = false;
+    }
+
+    private convertToColor3(color: string) {
+        if (color === "" || color === "transparent" || color === "Transparent") {
+            return new Color4(0, 0, 0, 0);
+        }
+
+        if (color.substring(0, 1) !== "#" || color.length !== 7) {
+            let d = document.createElement("div");
+            d.style.color = color;
+            document.body.append(d);
+            let rgb = window.getComputedStyle(d).color;
+            document.body.removeChild(d);
+
+
+            let rgbArray = rgb.substring(4, rgb.length - 1)
+                .replace(/ /g, '')
+                .split(',');
+
+            return new Color3(parseInt(rgbArray[0]) / 255, parseInt(rgbArray[1]) / 255, parseInt(rgbArray[2]) / 255);
+        }
+
+        var r = parseInt(color.substring(1, 3), 16);
+        var g = parseInt(color.substring(3, 5), 16);
+        var b = parseInt(color.substring(5, 7), 16);
+
+        return Color3.FromInts(r, g, b);

```

Comment:
Color3 has `FromHexString` to do exactly what you are doing here (parsing a hex string)